### PR TITLE
remove unused functions O to R

### PR DESCRIPTION
**Pull request by CasparKielwein**
Removes unused code in directories starting with Letters O to R, as described in #1955 .

Pretty much analogue to #1952

### CODE REVIEW

File: code/object/objectdock.cpp [hunk 1]
```diff
@@ -84,17 +83,6 @@ int dock_count_direct_docked_objects(object *objp)
 	return dock_count_instances(objp);
 }
 
-int dock_count_total_docked_objects(object *objp)
```
Comment (Goober5000) [inline code/object/objectdock.cpp]: Keep this function.

File: code/object/objectdock.cpp [hunk 2]
```diff
@@ -504,12 +492,6 @@ void dock_move_docked_children_tree(object *objp, object *parent_objp)
 // helper functions
 // ----------------
 
-void dock_count_total_docked_objects_helper(object * /*objp*/, dock_function_info *infop)
```
Comment (Goober5000) [inline code/object/objectdock.cpp]: And this.

File: code/object/objectdock.h [hunk 3]
```diff
@@ -117,7 +114,7 @@ float dock_calc_docked_fspeed(object *objp);
 // calculate the overall speed of the entire docked mass
 float dock_calc_docked_speed(object *objp);
 
-// �berfunction for evaluating all objects that could possibly be docked to objp.  This will
```
Comment (Goober5000) [inline code/object/objectdock.h]: Careful of the character encoding...

File: code/object/waypoint.cpp [hunk 4]
```diff
@@ -339,23 +339,6 @@ int find_index_of_waypoint_list(waypoint_list *wp_list)
 	return -1;
 }
 
-int find_index_of_waypoint(waypoint_list *wp_list, waypoint *wpt)
```
Comment (Goober5000) [inline code/object/waypoint.cpp]: Keep this function too.

File: code/particle/ParticleSource.cpp [hunk 5]
```diff
@@ -91,17 +91,6 @@ void SourceOrigin::applyToParticleInfo(particle_info& info, bool allowRelative)
 	}
 }
 
-vec3d SourceOrigin::getVelocity() const {
```
Comment (asarium) [inline code/particle/ParticleSource.cpp]: This should be kept in case a future particle effect uses it.

File: code/particle/ParticleSource.cpp [hunk 6]
```diff
@@ -252,26 +241,6 @@ bool SourceTiming::isFinished() const {
 	return timestamp_elapsed(m_endTimestamp) != 0;
 }
 
-float SourceTiming::getLifeTimeProgress() const {
```
Comment (asarium) [inline code/particle/ParticleSource.cpp]: Same here.

File: code/particle/ParticleSource.h [hunk 7]
```diff
@@ -71,8 +71,6 @@ class SourceOrigin {
 
 	inline SourceOriginType getType() const { return m_originType; }
 
-	inline object* getObjectHost() const { return m_origin.m_object.objp; }
```
Comment (asarium) [inline code/particle/ParticleSource.h]: Same here.

**Pull request merged**